utils/src/dense_polynomial.rs

```diff
@@ -57,6 +57,10 @@ impl<F: Field> ExtendedDensePolynomial<F> for DensePolynomial<F> {
             chunk_polys.push(DensePolynomial::from_coefficients_vec(vec![]));
         }
 
+        // Ensuring that the number of chunks is the one requested, following
+        // trait documentation
+        assert_eq!(chunk_polys.len(), num_chunks);
```

Comment:
No, it is not. See the commit message content.

Reply:
On line 51 you push some `N` elements, and then on line 57 you push `num_chunks - N` elements, so in the end you will have `N` elements anyway. I don't see how it's related to line 50 with vector initiation, you could've had `vec![]` there and it would've worked semantically in the same way. Am I missing smth?